gsy_framework/read_user_profile.py

```diff
@@ -385,3 +389,47 @@ def read_and_convert_identity_profile_to_float(profile, current_timestamp: DateT
 
 def _convert_identity_profile_to_float(profile, _current_timestamp: DateTime = None):
     return {k: float(v) for k, v in profile.items()}
+
+
+def _generate_time_slots(
+        slot_length: timedelta, sim_duration: timedelta, start_date: datetime
+) -> Generator:
+    return (start_date + timedelta(seconds=slot_length.total_seconds() * time_diff_count)
+            for time_diff_count in range(
+        int(sim_duration.total_seconds() / slot_length.total_seconds())))
+
+
+def _get_from_profile(profile: Dict[datetime, float], key: datetime) -> float:
+    try:
+        return profile[key]
+    except KeyError as ex:
+        raise GSyProfileException(f"The input profile does not contain the key {key}") from ex
+
+
+def resample_hourly_energy_profile(
+        input_profile: Dict[datetime, float], slot_length: timedelta, sim_duration: timedelta,
+        start_date: datetime) -> Dict[datetime, float]:
+    """Resample hourly energy profile in order to fit to the set slot_length."""
+    slot_length_minutes = slot_length.total_seconds() / 60
+    if slot_length_minutes < 60:
+        if 60 % slot_length_minutes != 0:
+            raise GSyProfileException("slot_length is not division of 1 hour")
```

Comment:
I think that it is more explicit if you use the term "exact division" in order to denote that only divisors of 60 are allowed as slot length minute values.

Reply:
Thanks a lot! My limited english was the reason for this formulation. Will change it like you propose